modules/rosa-cluster-hcp/main.tf

```diff
@@ -0,0 +1,135 @@
+    oidc_config_id       = var.oidc_config_id
+  }
+  aws_account_arn = var.aws_account_arn == null ? data.aws_caller_identity.current[0].arn : var.aws_account_arn
+}
+
+resource "rhcs_cluster_rosa_hcp" "rosa_hcp_cluster" {
+  name           = var.cluster_name
+  version        = var.openshift_version
+  upgrade_acknowledgements_for = var.upgrade_acknowledgements_for
+  private          = var.private
+  properties = merge(
+    {
+      rosa_creator_arn = local.aws_account_arn
+    },
+    var.properties
+  )
+  cloud_region   = var.aws_region == null ? data.aws_region.current[0].name : var.aws_region
+  aws_account_id = local.aws_account_id
+  aws_billing_account_id = var.aws_billing_account_id == null || var.aws_billing_account_id == "" ? (
+    data.aws_caller_identity.current[0].account_id
+  ) : (var.aws_billing_account_id)
+  sts            = local.sts_roles
+  tags = var.tags
+  availability_zones = length(var.aws_availability_zones) > 0 ? (
+    var.aws_availability_zones
+    ) : (
+    length(var.aws_subnet_ids) > 0 ? (
+      distinct(data.aws_subnet.provided_subnet[*].availability_zone)
+      ) : (
+      slice(data.aws_availability_zones.available[0].names, 0, 1)
+    )
+  )
+  replicas       = var.replicas
+  aws_subnet_ids = var.aws_subnet_ids
+  compute_machine_type = var.compute_machine_type
+
+  machine_cidr         = var.machine_cidr
+  service_cidr         = var.service_cidr
+  pod_cidr             = var.pod_cidr
+  host_prefix          = var.host_prefix
+  proxy = var.http_proxy != null || var.https_proxy != null || var.no_proxy != null || var.additional_trust_bundle != null ? (
+    {
+      http_proxy              = var.http_proxy
+      https_proxy             = var.https_proxy
+      no_proxy                = var.no_proxy
+      additional_trust_bundle = var.additional_trust_bundle
+    }
+    ) : (
+    null
+  )
+  etcd_encryption = var.etcd_encryption
+  etcd_kms_key_arn = var.etcd_kms_key_arn
+  kms_key_arn = var.kms_key_arn
+
+  wait_for_create_complete = var.wait_for_create_complete
+  wait_for_std_compute_nodes_complete = var.wait_for_std_compute_nodes_complete
+  disable_waiting_in_destroy   = var.disable_waiting_in_destroy
+  destroy_timeout              = var.destroy_timeout
+
+  lifecycle {
+    precondition {
+      condition = (
+        !(var.installer_role_arn != null && var.support_role_arn != null && var.worker_role_arn != null)
+        &&
+        var.account_role_prefix == null
+      ) == false
+      error_message = "Either provide the \"account_role_prefix\" or specify all ARNs for account roles (\"installer_role_arn\", \"support_role_arn\", \"controlplane_role_arn\", \"worker_role_arn\")."
+    }
+  }
+}
+
+resource "rhcs_hcp_cluster_autoscaler" "cluster_autoscaler" {
+  count = var.autoscaling_enabled == true ? 1 : 0
+
+  cluster                       = rhcs_cluster_rosa_hcp.rosa_hcp_cluster.id
+  max_pod_grace_period          = var.autoscaler_max_pod_grace_period
```

Comment:
Both `cluster_autoscaler` and `default_ingress` missing many variables options
Please compare to the classic module

Reply:
Those are not supported on HCP